command/agent/sink/sink.go

```diff
@@ -111,8 +113,12 @@ func (ss *SinkServer) Run(ctx context.Context, incoming chan string, sinks []*Si
 		case <-ctx.Done():
 			return nil
 
-		case token := <-incoming:
+		case ahOutput := <-incoming:
 			if len(sinks) > 0 {
+				token := ahOutput.Token
+				if ss.client != nil {
+					ss.client.WithRequestCallbacks(api.RequireState(ahOutput.VaultIndex))
```

Comment:
Isn't this a no-op?

Reply:
It's not - if the client is non-nil, we add the index returned by the auto-auth channel to the required state for the client. This does work, for the record.

Reply:
Hmm, wait, I might have been wrong. Investigating...